vulture_os/toolkit/api_parser/armis_centrix/armis_centrix.py

```diff
@@ -0,0 +1,259 @@
+                    )
+            except exceptions.ReadTimeout:
+                self.evt_stop.wait(10.0)
+                logger.info(f"[{__parser__}][execute_query] :: Encounters a ReadTimeout, sleeping {timeout} seconds and retry (retries left: {retry}/3)", extra={'frontend': str(self.frontend)})
+                continue
+            if response.status_code == 401:
+                resp = response.json()
+                # maybe a bit too much, this handle the case where token had expired or is invalid "silently" for an unknown reason
+                if resp.get("message", "") == "Expired access token.": 
+                    logger.info(f"[{__parser__}][execute_query] :: Resets expired access token {self.armis_token} - expired_at ({self.armis_token_expire_at})", extra={'frontend': str(self.frontend)})
+                    self.armis_token = "" # resets expired token
+                    self.login()
+                    continue
+                elif resp.get("message", "") == "Invalid access token.":
+                    logger.info(f"[{__parser__}][execute_query] :: Resets invalid access token {self.armis_token} - expiration ({self.armis_token_expire_at})", extra={'frontend': str(self.frontend)})
+                    self.armis_token = "" # resets expired token
+                    self.login()
+                    continue
+            break  # no error we can break
+
+        response.raise_for_status()
+        try:
+            return response.json()
+        except Exception as e:
+            logger.error(f"[{__parser__}][execute_query] :: Server awnsers with an invalid json response : status ({response.status_code}) - response ({response.content})")
+            raise Exception(e)
+
+
+    def get_logs(self, kind: str, since: datetime, to: datetime, test: bool = False) -> int|list:
+        logs = []
+
+        try:
+            since = since.strftime("%Y-%m-%dT%H:%M:%S")
+            to    =    to.strftime("%Y-%m-%dT%H:%M:%S")
+
+            url = f"{self.armis_host}{self.SEARCH_URL}"
+            if kind == "alerts":
+                url += "?aql=" + quote(f'in:alerts after:{since} before:{to} orderBy:(time asc)')
+            elif kind == "activity":
+                url += "?aql=" + quote(f'in:activity after:{since} before:{to} orderBy:(time asc)')
+            params = {
+                "length": self.LIMIT, # page size
+                "includeSample": "false",
+                "includeTotal": "false"
+            }
+            # get the first page of logs
+            resp = self.execute_query(
+                url=url,
+                method="GET",
+                query=params
+            )
+            logs = resp["data"]["results"]
+            totalToRetrieve = resp["data"]["total"]
+
+            if test:
+                return logs  # in case of test function exection we want to return the logs as is without paginating
```

Comment:
this is ugly AF: the function returns 2 kinds of data depending on this parameter -> this is not clean